src/api/MessageEvents.ts

```diff
@@ -46,16 +46,20 @@ const editListeners = new Set<EditListener>();
 export function _handlePreSend(channelId: string, messageObj: MessageObject, extra: any) {
     for (const listener of sendListeners) {
         try {
-            listener(channelId, messageObj, extra);
-        } catch (e) { MessageEventsLogger.error(`MessageSendHandler: Listener encoutered an unknown error. (${e})`); }
+            const result = listener(channelId, messageObj, extra);
+            if (result && result.cancel) {
```

Comment:
```suggestion
            if (result && result.cancel === true) {
```
just to make it more explicit